lib/components/DateTime/TimeInput.tsx

```diff
@@ -254,6 +201,22 @@ export class TimeInput extends React.Component<TimeInputProps, TimeInputState> {
         const seconds = this.state.seconds < 10 ? `0${this.state.seconds}` : this.state.seconds;
         const inputClassName = css('time-input', {'error': this.props.error}, this.props.inputClassName);
 
+        const hoursList = [];
```

Comment:
why did you move on the render rather then having it created one time?

Reply:
or, instead of creating the intermediate hoursList array, store the jsx elements directly: optionMap(option, this.props.attr.hourOption). That way, we'll preserve existing behavior of only looping over the hours once.

Reply:
I did the approach @vishwam suggested